rllib/examples/random_env.py

```diff
@@ -0,0 +1,69 @@
+"""
+Example of a custom gym environment and model. Run this for a demo.
+
+This example shows:
+  - using a custom environment
+  - using a custom model
+  - using Tune for grid search
+
+You can visualize experiment results in ~/ray_results using TensorBoard.
+"""
+
+import gym
+from gym.spaces import Tuple, Discrete
+import numpy as np
+
+from ray.rllib.agents.ppo import PPOTrainer
+from ray.rllib.utils import try_import_tf
+
+tf = try_import_tf()
+
+
+class RandomEnv(gym.Env):
```

Comment:
Should this be a test? Btw, we can also add an entry to `test_supported_spaces.py` which historically has been used to catch these sort of bugs.

Reply:
Yes, would be useful to add as a test.
I'll add both (Radom-env and supported-spaces) as tests (to Jenkins AND the new BAZEL branch).

Reply:
Hmm, we do check the support for tuple spaces for PPO, but not with the special config combination that caused this issue. Yeah, I was thinking about that last night: We should ideally expand this idea to more settings (add more loops): obs-space, action-space, framework, algo, lstm (yes/no), num_workers (0, 1, or 2), etc.. That would increase our test coverage a lot.